packages/ts-transform/src/utils/css-transform.tsx

```diff
@@ -0,0 +1,75 @@
+import postcss, { plugin } from 'postcss';
+import autoprefixer from 'autoprefixer';
+import cssnano from 'cssnano-preset-default';
+import nested from 'postcss-nested';
+import whitespace from 'postcss-normalize-whitespace';
+
+const minify = () => {
+  const preset = cssnano();
+  const asyncPluginsToExclude = ['postcss-svgo'];
+
+  return preset.plugins
+    .map(([creator]: any) => {
+      // replicate the `initializePlugin` behavior from https://github.com/cssnano/cssnano/blob/a566cc5/packages/cssnano/src/index.js#L8
+      return creator();
+    })
+    .filter((plugin: any) => {
+      return !asyncPluginsToExclude.includes(plugin.postcssPlugin);
+    });
+};
+
+const topLevelPseudos = plugin('top-level-pseudos', () => {
```

Comment:
I dont think this covers all cases yet:
- `.cls{span{:hover{}}}`
- `.cls{@media{:hover{}}}`
- `.cls{@supports{:hover{}}}`

Reply:
hey thanks for jumping in

does stylis (now your emotion plugin) transform these to

```
span.cls:hover {}
@media { .cls:hover { } }
@supports { .cls:hover { } }
```

i'll write some tests 😄

Reply:
gottem a539c05

Reply:
https://github.com/emotion-js/emotion/blob/5e39d81bf9177e68e78d23bd869c1592341d9768/packages/css/test/selectivity.test.js#L122

> span.cls:hover {} // <-- this one doesn't seem right to me tbh unless it was `&span` from the consumer

Should be:
```css
.cls span:hover {}
```

The "rule" how it works is actually pretty simple - orphaned leading pseudo always acts as `&:pseudo`. So the difference with and without this plugin is a single whitespace (or lack of it) between "parent" selector part (`&`) and the current selector containing pseudo as the first item.

Reply:
ah interesting - does this behaviour add it for any depth of nested orphaned leading pseudos?

Reply:
yes, the behavior is the same for all depths

Reply:
pushed up a fix. does it look good to you? 😄

Reply:
some of my test cases are kinda dumb/weird now 😅 i've added them because i was falling into some edge cases with my prior approach (before I've implemented `.parent` on stylis nodes which just made the final plugin **way** easier), wasn't feeling like removing them though

Reply:
wheres your plugin located? maybe i should just look at it 😺

Reply:
https://github.com/emotion-js/emotion/blob/5e39d81bf9177e68e78d23bd869c1592341d9768/packages/cache/src/stylis-plugins.js#L49